### PR TITLE
🌱 Install golangci-lint directly instead of building it using go-modules

### PR DESCRIPTION
**Pull request by kashifest**
as a benefit we get rid of the hashicorp dependencies from `hack/tools/go.mod`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -331,9 +340,10 @@ go-version: ## Print the go version we use to compile our binaries and images
 ## --------------------------------------
 
 .PHONY: clean
-clean: ## Remove all temporary files and folders
+clean: ## Remove all temporary files,folders and tools
```
Comment (tuminoid) [inline Makefile:343]:
```suggestion
clean: ## Remove all temporary files, directories and tools
```
Missing space + folder is something on a desktop

**Review by tuminoid — COMMENTED** [review]
One tiny nit.

**Comment by maxrantil** [comment]
Whaow! That's a lot of cleaning up lines :) LGTM

File: Makefile [hunk 2]
```diff
@@ -120,13 +128,14 @@ test-e2e: $(GINKGO) ## Run the end-to-end tests
 .PHONY: linters
 linters: lint generate-check fmt-check
 
-tools/bin/golangci-lint: hack/tools/go.mod
-	cd hack/tools; go build -o $(abspath $@) github.com/golangci/golangci-lint/cmd/golangci-lint
+$(GOLANGCI_LINT):
```
Comment (tuminoid) [inline Makefile:131]: Can you add phony target for $(GOLANGCI_LINT_BIN) so it is easier to call as is? Now you need to call `make $(pwd)/hack/tools/bin/golangci-lint` to make it.

Comment (tuminoid) [reply Makefile:131]: I can also add it in follow-up, if not feeling like it.

Comment (kashifest) [reply Makefile:131]: I will add it here in a while

Comment (kashifest) [reply Makefile:131]: Done

**Review by lentzi90 — COMMENTED** [review]
/lgtm

File: Makefile (hunk 2 above)
Comment (tuminoid) [reply Makefile:131]: Thanks. 

/lgtm

**Comment by honza** [comment]
This is great.

/approve

**Pull request merged**